File: api/services/nearby_airports.py
"""Static mapping of airports to nearby alternatives with drive times."""

# Maps airport code -> list of (nearby_code, drive_time_minutes)
NEARBY_AIRPORTS: dict[str, list[tuple[str, int]]] = {
    "YEG": [("YYC", 180)],
    "YYC": [("YEG", 180)],
    "YVR": [("SEA", 180), ("BLI", 150)],
    "YYZ": [("BUF", 120), ("YHM", 60)],
    "YUL": [("YOW", 120)],
    "YOW": [("YUL", 120), ("YYZ", 270)],
    "JFK": [("EWR", 40), ("LGA", 30)],
    "EWR": [("JFK", 40), ("LGA", 30)],
    "LGA": [("JFK", 30), ("EWR", 40)],
    "SFO": [("OAK", 30), ("SJC", 45)],
    "LAX": [("BUR", 30), ("SNA", 45), ("ONT", 50)],
    "IAD": [("DCA", 45), ("BWI", 60)],
    "DCA": [("IAD", 45), ("BWI", 40)],
    "ORD": [("MDW", 40)],
    "DFW": [("DAL", 30)],
    "MIA": [("FLL", 40)],
    "BOS": [("PVD", 60), ("MHT", 60)],
    "SEA": [("YVR", 180)],
    "DEN": [("COS", 80)],
    "PHX": [("TUS", 100)],
}


def get_nearby(code: str) -> list[tuple[str, int]]:
    """Return list of (airport_code, drive_time_minutes) for nearby airports."""
    return NEARBY_AIRPORTS.get(code.upper(), [])

File: app/src/lib/nearby-airports.ts
export const NEARBY_AIRPORTS: Record<
  string,
  { code: string; city: string; drive_minutes: number }[]
> = {
  YEG: [{ code: "YYC", city: "Calgary", drive_minutes: 180 }],
  YYC: [{ code: "YEG", city: "Edmonton", drive_minutes: 180 }],
  YVR: [{ code: "SEA", city: "Seattle", drive_minutes: 180 }],
  YYZ: [
    { code: "BUF", city: "Buffalo", drive_minutes: 120 },
    { code: "YHM", city: "Hamilton", drive_minutes: 60 },
  ],
  JFK: [
    { code: "EWR", city: "Newark", drive_minutes: 40 },
    { code: "LGA", city: "LaGuardia", drive_minutes: 30 },
  ],
  SFO: [
    { code: "OAK", city: "Oakland", drive_minutes: 30 },
    { code: "SJC", city: "San Jose", drive_minutes: 45 },
  ],
  LAX: [
    { code: "BUR", city: "Burbank", drive_minutes: 30 },
    { code: "SNA", city: "Santa Ana", drive_minutes: 45 },
  ],
  IAD: [
    { code: "DCA", city: "Reagan National", drive_minutes: 45 },
    { code: "BWI", city: "Baltimore", drive_minutes: 60 },
  ],
  ORD: [{ code: "MDW", city: "Midway", drive_minutes: 40 }],
  MIA: [{ code: "FLL", city: "Fort Lauderdale", drive_minutes: 40 }],
  SEA: [{ code: "YVR", city: "Vancouver", drive_minutes: 180 }],
};
